### PR TITLE
Improve context handling for Post Types.

### PR DESCRIPTION
**Pull request by bordoni**
I noticed when doing some canonical redirect work for B20.02, I realized that some of our variables were not looking for all the correct places to determine the Post Type related variables.

### CODE REVIEW

File: src/Tribe/Context/locations.php [hunk 1]
```diff
@@ -76,6 +76,25 @@
 	],
 	'post_type' => [
 		'read' => [
+			Tribe__Context::FUNC        => static function() {
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 2b44e6c0ae8560c1570663b3587b0d5f</sub>

File: src/Tribe/Context/locations.php [hunk 2]
```diff
@@ -76,6 +76,25 @@
...
+				$post_type_objs = get_post_types( [
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: 3488e94c3542c8cdb732b858968dbec1</sub>

File: src/Tribe/Context/locations.php [hunk 3]
```diff
@@ -76,6 +76,25 @@
...
+					'public' => true,
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :warning: Array double arrow not aligned correctly; expected 3 space(s) between "'public'" and double arrow, but found 1.
* :no_entry_sign: TRUE, FALSE and NULL must be uppercase; expected "TRUE" but found "true"

<sub>hash: e9ac134d0d0ab3414abd88998191868d</sub>

File: src/Tribe/Context/locations.php [hunk 4]
```diff
@@ -76,6 +76,25 @@
...
+					'_builtin' => false,
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: TRUE, FALSE and NULL must be uppercase; expected "FALSE" but found "false"

<sub>hash: 2f81754420dd200a2788002ef87894b9</sub>

File: src/Tribe/Context/locations.php [hunk 5]
```diff
@@ -76,6 +76,25 @@
...
+				], 'objects' );
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: df1dfb55d90e5dd421f34f755c8e43b3</sub>

File: src/Tribe/Context/locations.php [hunk 6]
```diff
@@ -76,6 +76,25 @@
...
+
+				foreach( $post_type_objs as $post_type ) {
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: There must be a space between foreach and the first parenthesis '('
* :no_entry_sign: Space after opening control structure is required
* :no_entry_sign: No space before opening parenthesis is prohibited
* :no_entry_sign: Expected 1 space(s) after FOREACH keyword; 0 found

<sub>hash: bc722711b2c223ec92f8ea970410465f</sub>

File: src/Tribe/Context/locations.php [hunk 7]
```diff
@@ -76,6 +76,25 @@
...
+					if ( empty( $post_type->query_var ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 9459cf1191e254c7bb981d77d355ed0d</sub>

File: src/Tribe/Context/locations.php [hunk 8]
```diff
@@ -76,6 +76,25 @@
...
+						continue;
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: ff617c134127af27459182ad52911af6</sub>

File: src/Tribe/Context/locations.php [hunk 9]
```diff
@@ -76,6 +76,25 @@
...
+					}
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 76565195498bc29098cfb69af17cb774</sub>

File: src/Tribe/Context/locations.php [hunk 10]
```diff
@@ -76,6 +76,25 @@
...
+
+					$url_value = tribe_get_request_var( $post_type->query_var, false );
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: * :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed
* :no_entry_sign: TRUE, FALSE and NULL must be uppercase; expected "FALSE" but found "false"
* :warning: Line exceeds 80 characters; contains 87 characters

<sub>hash: 860681149651c8f922819dcc1b5f2e7a</sub>

File: src/Tribe/Context/locations.php [hunk 11]
```diff
@@ -76,6 +76,25 @@
...
+					if ( empty( $url_value ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 249d202c1ec6611621de473268e89260</sub>

File: src/Tribe/Context/locations.php [hunk 12]
```diff
@@ -76,6 +76,25 @@
...
+						continue;
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 4b89db927e36ff17294b69ede360b023</sub>

File: src/Tribe/Context/locations.php [hunk 13]
```diff
@@ -76,6 +76,25 @@
...
+					}
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 8124a5e566f94a6a20b64c5b8455b4a6</sub>

File: src/Tribe/Context/locations.php [hunk 14]
```diff
@@ -76,6 +76,25 @@
...
+
+					return $post_type->name;
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 39a7439c77ea4b0fd3c081f4a3d1a769</sub>

File: src/Tribe/Context/locations.php [hunk 15]
```diff
@@ -76,6 +76,25 @@
...
+				}
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 0ed46ed13bc4e474e76da70bfa9cf3c2</sub>

File: src/Tribe/Context/locations.php [hunk 16]
```diff
@@ -76,6 +76,25 @@
...
+			},
```
Comment (tr1b0t) [inline src/Tribe/Context/locations.php]: :no_entry_sign: Spaces must be used to indent lines; tabs are not allowed

<sub>hash: 79798693120e7e8934f8a2e46aa1e26a</sub>

File: src/Tribe/Context/locations.php (hunk 15 above)
Comment (lucatume) [inline src/Tribe/Context/locations.php]: For the context to work correctly, here the code should return `Tribe__Context::NOT_FOUND` when the value we're trying to fetch is not found.

**Pull request merged**